File: fe/src/services/api/routes/laporan.route.js
import api from "../axios";
export const laporanAPI = {
  getAll: (params) => api.get("/laporan", { params }),
  getMilikSaya: (params) => api.get("/laporan/my-laporan", { params }),
  getById: (id) => api.get(`/laporan/${id}`),
  create: (data, fotoBuktiFiles) => {
    if (fotoBuktiFiles && fotoBuktiFiles.foto_bukti_urls) {
      const formData = new FormData();
      fotoBuktiFiles.foto_bukti_urls.forEach((file) => {
        formData.append("foto_bukti_urls", file);
      });
      Object.entries(data).forEach(([key, value]) => {
        if (value !== "" && value !== null && value !== undefined) {
          formData.append(key, value);
        }
      });
      return api.post("/laporan", formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.post("/laporan", data);
  },
  update: (id, data, fotoBuktiFiles = null) => {
    if (fotoBuktiFiles || (data && Array.isArray(data.existing_foto_bukti))) {
      const formData = new FormData();
      if (fotoBuktiFiles && fotoBuktiFiles.foto_bukti_urls) {
        fotoBuktiFiles.foto_bukti_urls.forEach((file) => {
          formData.append("foto_bukti_urls", file);
        });
      }
      Object.entries(data).forEach(([key, value]) => {
        if (value !== "" && value !== null && value !== undefined) {
          if (Array.isArray(value)) {
            value.forEach((val) => formData.append(key, val));
          } else {
            formData.append(key, value);
          }
        }
      });
      return api.put(`/laporan/${id}`, formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.put(`/laporan/${id}`, data);
  },
  updateStatus: (id, data) => api.patch(`/laporan/${id}/status`, data),
  delete: (id) => api.delete(`/laporan/${id}`),

  // Tindak Lanjut
  getTindakLanjut: (laporanId) =>
    api.get(`/laporan/${laporanId}/tindak-lanjut`),
  createTindakLanjut: (laporanId, data, fotoFiles) => {
    if (fotoFiles) {
      const formData = new FormData();
      if (fotoFiles.foto_sebelum) {
        fotoFiles.foto_sebelum.forEach((file) =>
          formData.append("foto_sebelum_tindakan_urls", file),
        );
      }
      if (fotoFiles.foto_setelah) {
        fotoFiles.foto_setelah.forEach((file) =>
          formData.append("foto_setelah_tindakan_urls", file),
        );
      }
      Object.entries(data).forEach(([key, value]) => {
        if (value !== "" && value !== null && value !== undefined) {
          formData.append(key, value);
        }
      });
      return api.post(`/laporan/${laporanId}/tindak-lanjut`, formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.post(`/laporan/${laporanId}/tindak-lanjut`, data);
  },
};

File: be/app/api/routes/laporan_route.py
"""Laporan routes"""
from flask import Blueprint
from app.api.controllers import laporan_controller

laporan_bp = Blueprint('laporan', __name__, url_prefix='/api/laporan')


@laporan_bp.route('', methods=['GET'])
def get_all():
    return laporan_controller.get_all()


@laporan_bp.route('/<item_id>', methods=['GET'])
def get_one(item_id):
    return laporan_controller.get_one(item_id)


@laporan_bp.route('', methods=['POST'])
def create():
    return laporan_controller.create()


@laporan_bp.route('/<item_id>', methods=['PUT', 'PATCH'])
def update(item_id):
    return laporan_controller.update(item_id)


@laporan_bp.route('/<item_id>/status', methods=['PATCH'])
def update_status(item_id):
    return laporan_controller.update_status(item_id)


@laporan_bp.route('/<item_id>', methods=['DELETE'])
def delete(item_id):
    return laporan_controller.delete(item_id)


# Tindak Lanjut (nested under laporan)
@laporan_bp.route('/<laporan_id>/tindak-lanjut', methods=['GET'])
def get_tindak_lanjut(laporan_id):
    return laporan_controller.get_tindak_lanjut(laporan_id)


@laporan_bp.route('/<laporan_id>/tindak-lanjut', methods=['POST'])
def create_tindak_lanjut(laporan_id):
    return laporan_controller.create_tindak_lanjut(laporan_id)


@laporan_bp.route('/my-laporan', methods=['GET'])
def my_laporan():
    return laporan_controller.my_laporan()
